### PR TITLE
samples: remove unnecessary check for flags cell in gpios

### LINKED ISSUE #27803: samples: update to support new devicetree flag defaults
The merge of #27076 made the complex steps to accommodate GPIO specifiers that lack a flags cell, i.e.:
```
#define LED0    DT_GPIO_LABEL(LED0_NODE, gpios)
#define PIN     DT_GPIO_PIN(LED0_NODE, gpios)
#if DT_PHA_HAS_CELL(LED0_NODE, gpios, flags)
#define FLAGS   DT_GPIO_FLAGS(LED0_NODE, gpios)
#endif
```

All in-tree samples should be updated to remove this unnecessary test, as it is being replicated in new code like #26002.

### PR DESCRIPTION
**Pull request by pabigot**
The devicetree helper macro now provides zero when the cell is not present.  Update the samples so we don't propagate the outdated idiom.

Closes #27803

### CODE REVIEW

File: samples/sensor/fxos8700-hid/src/main.c [hunk 1]
```diff
@@ -15,38 +15,22 @@
...
-#if DT_NODE_HAS_PROP(DT_ALIAS(sw1), gpios)
-#define PORT1	DT_GPIO_LABEL(DT_ALIAS(sw1), gpios)
-#endif
-
-#if DT_PHA_HAS_CELL(DT_ALIAS(sw1), gpios, flags)
-#define PIN1_FLAGS DT_GPIO_FLAGS(DT_ALIAS(sw1), gpios)
+#if DT_NODE_HAS_STATUS(SW1_NODE, okay)
+#define PORT1	DT_GPIO_LABEL(SW1_NODE, gpios)
+#define PIN1	DT_GPIO_PIN(SW1_NODE, gpios)
+#define PIN1_FLAGS DT_GPIO_FLAGS(SW1_NODE, gpios)
```
Comment (dleach02) [inline samples/sensor/fxos8700-hid/src/main.c:33]: minor nit but the code block for SW0_NODE you aligned the define with spaces but this one you didn't

**Review by dleach02 — APPROVED** [review]
Other than my nit this is much cleaner!

File: samples/sensor/fxos8700-hid/src/main.c (hunk 1 above)
Comment (pabigot) [reply samples/sensor/fxos8700-hid/src/main.c]: Turns out this file aligns with tabs.  Now its consistent.

**Review by mbolivar-nordic — APPROVED** [review]
Thanks for this.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**